bluemira/base/components.py

```diff
@@ -269,7 +281,9 @@ def filter_(node, properties):
             ]
             return tuple(map(list, zip(*node_properties)))
 
-    def _get_thing(self, filter_: Callable, first: bool, full_tree: bool):
+    def _get_thing(
+        self, filter_: Union[Callable, None], first: bool, full_tree: bool
+    ) -> Union[Component, Iterable[Component], None]:
```

Comment:
Again, we should be specific about the return type (which I think is `Tuple` here?)